stdlib/src/builtin/simd.mojo

```diff
@@ -2192,6 +2192,32 @@ struct SIMD[type: DType, size: Int = simdwidthof[type]()](
             "llvm.vector.reduce.or", Scalar[DType.bool], has_side_effect=False
         ](self)
 
+    @always_inline
+    fn _reduce_all(self) -> Bool:
+        """Returns `True` if **all** elements in this vector are non-zero.
+
```

Comment:
Can you please add a note here and in `_reduce_any` how these are different from their publilc counterparts (e.g. "unlike the public `reduce_all` this function is not constrained to boolean element type")

Reply:
i dont think there is a need to do that right now, were still not sure what will happen to it, and if `reduce_and`/`reduce_or` become bitwise operations, it will be mentioned in the docstring.